src/main/java/be/acara/events/service/OrderServiceImpl.java

```diff
@@ -0,0 +1,78 @@
+package be.acara.events.service;
+
+import be.acara.events.domain.CreateOrder;
+import be.acara.events.domain.Event;
+import be.acara.events.domain.Order;
+import be.acara.events.domain.User;
+import be.acara.events.exceptions.EventNotFoundException;
+import be.acara.events.exceptions.IdAlreadyExistsException;
+import be.acara.events.exceptions.IdNotFoundException;
+import be.acara.events.repository.OrderRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+
+@Service
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final EventService eventService;
+    private final UserService userService;
+
+    @Autowired
+    public OrderServiceImpl(OrderRepository orderRepository, EventService eventService, UserService userService) {
+        this.orderRepository = orderRepository;
+        this.eventService = eventService;
+        this.userService = userService;
+    }
+
+    @Override
+    public Order findById(Long id) {
+        return orderRepository.findById(id)
+                .orElseThrow(() -> new EventNotFoundException(String.format("Order with ID %d not found", id)));
+    }
+
+
+    @Override
+    public Order create(CreateOrder createOrder) {
+        Event event = eventService.findById(createOrder.getEventId());
+        String userName = SecurityContextHolder.getContext().getAuthentication().getName();
+        User user = userService.findByUsername(userName);
+        Order order = orderRepository.saveAndFlush(
+                Order.builder()
+                        .event(event)
+                        .user(user)
+                        .amountOfTickets(createOrder.getAmountOfTickets())
+                        .total(event.getPrice().multiply(new BigDecimal(createOrder.getAmountOfTickets())))
+                        .build());
+        eventService.addAttendee(event, user);
+        return order;
```

Comment:
move eventService.addAttendee(event, user) above Order order = orderRepsoitory.... and return the order directly. (return orderRepository....)

Keep in mind the other comment regarding eventService.addAttendee